src/main/webapp/app/modeling-assessment-editor/modeling-assessment-conflict/modeling-assessment-conflict.component.ts

```diff
@@ -50,6 +81,90 @@ export class ModelingAssessmentConflictComponent implements OnInit, AfterViewIni
+        );
+        if (elementAssessmentUpdate.credits !== originalElementAssessment.credits) {
+            this.updateCurrentState();
+        }
+        this.mergedFeedbacks = feedbacks;
+    }
+
+    onSave() {
+        this.modelingAssessmentService.saveAssessment(this.mergedFeedbacks, this.submissionId).subscribe(
+            result => {
+                this.jhiAlertService.success('modelingAssessmentEditor.messages.saveSuccessful');
+            },
+            error => this.jhiAlertService.error('modelingAssessmentEditor.messages.saveFailed'),
+        );
+    }
+
+    onSubmit() {
+        const escalatedConflicts: Conflict[] = [];
+        for (let i = 0; i < this.conflictResolutionStates.length; i++) {
+            if (this.conflictResolutionStates[i] === ConflictResolutionState.ESCALATED) {
+                escalatedConflicts.push(this.conflicts[i]);
+            }
+        }
+        this.modelingAssessmentService.escalateConflict(escalatedConflicts).subscribe(value => {
+            this.modelingAssessmentService.saveAssessment(this.mergedFeedbacks, this.submissionId, true).subscribe(
+                result => {
+                    this.jhiAlertService.success('modelingAssessmentEditor.messages.submitSuccessful');
+                    this.router.navigate(['modeling-exercise', this.modelingExercise.id, 'submissions', this.submissionId, 'assessment']);
+                },
+                error => {
+                    if (error.status === 409) {
+                        this.conflicts = error.error as Conflict[];
+                        this.conflicts.forEach((conflict: Conflict) => {
+                            this.modelingAssessmentService.convertResult(conflict.causingConflictingResult.result);
+                            conflict.resultsInConflict.forEach((conflictingResult: ConflictingResult) => this.modelingAssessmentService.convertResult(conflictingResult.result));
+                        });
+                        this.initComponent();
+                        this.jhiAlertService.clear();
+                        this.jhiAlertService.error('modelingAssessmentEditor.messages.submitFailedWithConflict');
+                    } else {
+                        this.jhiAlertService.clear();
+                        this.jhiAlertService.error('modelingAssessmentEditor.messages.submitFailed');
+                    }
+                },
+            );
+        });
+    }
+
+    updateFeedbackInMergedFeedback(elementIdToUpdate: string, elementIdToUpdateWith: string, sourceFeedbacks: Feedback[]) {
+        const feedbacks: Feedback[] = [];
+        const feedbackToUse = sourceFeedbacks.find((feedback: Feedback) => feedback.referenceId === elementIdToUpdateWith);
+        this.mergedFeedbacks.forEach(feedback => {
+            if (feedback.referenceId === elementIdToUpdate) {
+                feedback.credits = feedbackToUse.credits;
```

Comment:
feedbackToUse.credits could also cause the "property ... of undefined" issue.

Reply:
fixed